delta-sharing/server/src/auth.rs

```diff
@@ -1,37 +1,58 @@
-use std::sync::Arc;
+//! Authentication middleware for Delta Sharing server.
+
 use std::task::{Context, Poll};
 
 use axum::extract::Request;
 use axum::response::{IntoResponse, Response};
-use delta_sharing_core::DeltaRecipient;
-use delta_sharing_core::{Authenticator, Error as CoreError};
+use delta_sharing_core::{Authenticator, DeltaRecipient, Error as CoreError};
 use futures_util::{future::BoxFuture, FutureExt};
 use tower::{Layer, Service};
 
 use crate::error::{Error, Result};
 
+/// Authenticator that always marks the recipient as anonymous.
+#[derive(Clone)]
 pub struct AnonymousAuthenticator;
 
 impl Authenticator for AnonymousAuthenticator {
     type Request = Request;
     type Recipient = DeltaRecipient;
 
-    fn authenticate(&self, _: &Self::Request) -> Result<Self::Recipient, CoreError> {
+    fn authenticate(&self, _request: &Self::Request) -> Result<Self::Recipient, CoreError> {
```

Comment:
there is a slight nuance here and in practice there likely is no difference, but `_request` will create a variable binding that is dropped at the end of the scope, while for `_` this binding will never be created ...

Reply:
in the latter case it just tells the compiler not to complain about an unused variable.

Reply:
Today I learned! 😃

Reply:
Should be fixed now 🎉